app/src/server/api/routers/datasets.router.ts

```diff
@@ -82,10 +87,15 @@ export const datasetsRouter = createTRPCRouter({
       await prisma.dataset.update({
         where: { id: input.id },
         data: {
-          name: input.name,
+          name: input.updates.name,
+          enabledComparisonModels: input.updates.enabledComparisonModels,
         },
       });
 
+      if (input.updates.enabledComparisonModels) {
+        await startDatasetTestJobs(input.id);
```

Comment:
Can we run this in a background function? It took a surprisingly long time for the popup to clear for me.

Reply:
Agreed, I've noticed the same thing.

Reply:
Yeah just tried again with a 10k row dataset and we're 2 minutes in and the button is still spinning. Definitely feels broken.

Reply:
We're no longer calculating prunedInputTokens before creating these test entries, which dramatically speeds up the process.